### ISSUE TITLE
Question about the methodology

### ISSUE THREAD

**Issue by pg2455** [KimMeen/Time-LLM#143]
Great work!

I am wondering whether gradients are flowing through the llm_model in line 243 here: [https://github.com/KimMeen/Time-LLM/blob/main/models/TimeLLM.py#L243](https://github.com/KimMeen/Time-LLM/blob/main/models/TimeLLM.py#L243). 

The text says that the llm_model is frozen, so I cannot understand how the gradients might flow through it. Also, I don't see any provision to stop llm_model from updating (fine-tuning) itself. 

Any help will be highly appreciated. Thanks!

**Comment by AleksandrSim** [comment]
Hi, I’m not the author of the project, but this is how I understand the gradient updates in the model:

Regarding the flow through the llm_model in line 243: Although the model is frozen (meaning requires_grad is set to False for its parameters), gradients can still flow through the llm_model to update earlier layers, such as the ReprogrammingLayer. Freezing the llm_model prevents it from being updated during backpropagation, but it still passes gradients to the layers before it.

In short, while the llm_model itself won’t be updated, layers like the ReprogrammingLayer and other parts of the model that are not part of the llm_model will still receive gradient updates and will be optimized during training. Will be very glad to hear any other inside regarding the question and how the process there can be potentially optimized!

**Comment by pg2455** [comment]
Hi,  I found the lines of code that disables learning in LLM. Thank you very much for your response!
